rmw_opensplice_cpp/src/rmw_publish.cpp

```diff
@@ -67,10 +108,51 @@ rmw_ret_t
 rmw_publish_serialized_message(
   const rmw_publisher_t * publisher, const rmw_serialized_message_t * serialized_message)
 {
-  (void) publisher;
-  (void) serialized_message;
+  if (!publisher) {
+    RMW_SET_ERROR_MSG("publisher handle is null");
+    return RMW_RET_ERROR;
+  }
 
-  RMW_SET_ERROR_MSG("rmw_publish_serialized_message is not yet implemented");
-  return RMW_RET_ERROR;
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    publisher handle,
+    publisher->implementation_identifier, opensplice_cpp_identifier,
+    return RMW_RET_ERROR)
+
+  if (!serialized_message) {
+    RMW_SET_ERROR_MSG("ros serialized_message handle is null");
+    return RMW_RET_ERROR;
+  }
+
+  OpenSpliceStaticPublisherInfo * publisher_info =
+    static_cast<OpenSpliceStaticPublisherInfo *>(publisher->data);
+  if (!publisher_info) {
+    RMW_SET_ERROR_MSG("publisher info handle is null");
+    return RMW_RET_ERROR;
+  }
+
+  DDS::DataWriter * topic_writer = publisher_info->topic_writer;
+
+  const message_type_support_callbacks_t * callbacks = publisher_info->callbacks;
+  if (!callbacks) {
+    RMW_SET_ERROR_MSG("callbacks handle is null");
+    return RMW_RET_ERROR;
+  }
+
+  DDS::OpenSplice::FooCdrDataWriter cdr_writer(topic_writer);
+
+  DDS::CDRSample sample;
+
+  auto buffer = (unsigned char *)serialized_message->buffer;
+  auto length = serialized_message->buffer_length;
+
+  sample.blob = DDS::CDRBlob(length, (DDS::ULong)length, buffer, false);
```

Comment:
This line still results in a warning: https://ci.ros2.org/job/ci_windows/5553/warnings43Result/new/category.63474818/